go/vt/vtadmin/api.go

```diff
@@ -221,13 +221,42 @@ func (api *API) GetKeyspaces(ctx context.Context, req *vtadminpb.GetKeyspacesReq
 				return
 			}
 
-			m.Lock()
+			kss := make([]*vtadminpb.Keyspace, 0, len(resp.Keyspaces))
+
+			var (
+				kwg sync.WaitGroup
+				km  sync.Mutex
+			)
+
 			for _, ks := range resp.Keyspaces {
-				keyspaces = append(keyspaces, &vtadminpb.Keyspace{
-					Cluster:  c.ToProto(),
-					Keyspace: ks,
-				})
+				kwg.Add(1)
+
+				// Find all shards for each keyspace in the cluster, in parallel
+				go func(c *cluster.Cluster, ks *vtctldatapb.Keyspace) {
```

Comment:
I'm ambivalent about the nested goroutine. Well... my feeling is less ambivalence and more active dislike, haha. 

I considered pulling the goroutine into its own `getShardsForKeyspace` function, but that felt less readable. I also considered doing things serially (especially since on this tiny, local cluster, using goroutines adds 50ms somehow, haha) _but_ I figured I may as well, since I expect it'll be a significant improvement in a real deployment...? 

Anyway. Suggestions welcome.